lib/improver/nowcasting/pysteps_interfacing.py

```diff
@@ -0,0 +1,145 @@
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Functions to assist with interface to pysteps library
+"""
+
+import numpy as np
+
+from improver.utilities.load import load_cube
+from improver.utilities.spatial import check_if_grid_is_equal_area
+
+
+class PystepsImporter(object):
+    """
+    Class containing methods to import from IMPROVER NetCDF file or cube
+    and return data and metadata.  This is to allow use of the pysteps
+    plotting functions in investigations only, and should not be required for
+    any of the pysteps processing routines.
+
+    Reference:
+        https://pysteps.readthedocs.io/en/latest/pysteps_reference/
+        io.html#pysteps-io-importers
+    """
+    def __init__(self):
+        """
+        Set up some universally required metadata
+        """
+        self.metadata = {}
+        self.metadata['unit'] = 'mm/h'     # pysteps-acceptable units
+        self.metadata['accutime'] = 0      # accumulation period (0 for rates)
+        self.metadata['transform'] = None  # data transform
+
+    @staticmethod
+    def _extract_coord_properties(coord):
+        """Get grid spacing and coordinate extremes"""
+        coord.convert_units('m')
+        spacing = np.diff(coord.points)[0]
+        return spacing, min(coord.points), max(coord.points)
+
+    def _set_coord_metadata(self):
+        """Extract metadata from x- and y-coordinates and set in dict"""
+        xspacing, min_x, max_x = self._extract_coord_properties(
+            self.cube.coord(axis='x'))
+        self.metadata['xpixelsize'] = xspacing
+        self.metadata['x1'] = min_x
+        self.metadata['x2'] = max_x
+        yspacing, min_y, max_y = self._extract_coord_properties(
+            self.cube.coord(axis='y'))
+        self.metadata['ypixelsize'] = yspacing
+        self.metadata['y1'] = min_y
+        self.metadata['y2'] = max_y
+        self.metadata['yorigin'] = 'lower'
+
+    def _set_geodata(self):
+        """
+        Set projection and coordinate information in metadata dictionary
+        """
+        projdef = ""
+        crd_sys = self.cube.coord_system()
+        for k, v in crd_sys.as_cartopy_crs().proj4_params.items():
+            try:
+                projdef += "+{}={:.3f} ".format(k, float(v))
+            except ValueError:
+                projdef += "+{}={} ".format(k, v)
+        ellps = "WGS84"
+        projdef += "+ellps={}".format(ellps)
+        self.metadata["projection"] = projdef
+        self._set_coord_metadata()
+
+    def process_cube(self, precip_cube):
+        """
+        Read required data and metadata from cube
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Input precipitation rate cube with IMPROVER metadata
+
+        Returns:
+            precip_rate (np.ndarray):
+                2D array of precipitation rates in mm/h
+            metadata (dict):
+                Dictionary of metadata required by pysteps algorithms
+        """
+        if 'rate' not in precip_cube.name():
```

Comment:
Is this sufficient? I'm a bit surprised that if we are going to bother to check that this is a precipitation rate cube (in addition to the unit-conversion below) that we aren't checking for the string "precip" too. Are you thinking that we could do rain and snow separately?

Reply:
As discussed - we think there is no perfect way to do this, but this way is "good enough".